content/developer/reference/frontend/hooks.rst

```diff
@@ -282,3 +286,70 @@ API
           </t>
         </div>
       `;
+
+.. _frontend/hooks/useSpellCheck:
+
+useSpellCheck
+=============
+
+Location
+--------
+
+`@web/core/utils/hooks`
+
+Description
+-----------
+
+Activate the spellcheck state to an input or textarea on focus by a `t-ref="spellcheck"` in
+the current component. This state is then removed on blur, as well as the red outline, which
+improves readability of the content.
+
+The hook can also be used on any HTML element with the `contenteditable` attribute. To disable
+spellcheck completely on elements that might be enabled by the hook, set explicitly the
+`spellcheck` attribute as `false` on the element.
+
+In the following example, the spellcheck will be enabled on the first input, the textarea and
+the div with `contenteditable="true"`.
+
+.. code-block:: javascript
+
+    import { useSpellCheck } from "@web/core/utils/hooks";
+
+    class Comp {
+      setup() {
+        this.simpleRef = useSpellCheck();
+        this.customRef = useSpellCheck({ refName: "custom" });
+        this.nodeRef = useSpellCheck({ refName: "container" });
+      }
+      static template = "Comp";
+    }
+
+.. code-block:: xml
+
+    <t t-name="Comp" owl="1">
+      <input t-ref="spellcheck" type="text"/>
+      <textarea t-ref="custom"/>
+      <div t-ref="container">
+        <input type="text" spellcheck="false"/>
+        <div contenteditable="true"/>
+      </div>
+    </t>
```

Comment:
Use the `example` directive to highlight and clearly separate examples from the rest of the reference doc.
```suggestion
.. example::

   In the following example, the spellcheck will be enabled on the first input, the textarea and
   the div with `contenteditable="true"`.
   
   .. code-block:: javascript
   
       import { useSpellCheck } from "@web/core/utils/hooks";
       class Comp {
         setup() {
           this.simpleRef = useSpellCheck();
           this.customRef = useSpellCheck({ refName: "custom" });
           this.nodeRef = useSpellCheck({ refName: "container" });
         }
         static template = "Comp";
       }
   
   .. code-block:: xml
   
       <t t-name="Comp" owl="1">
         <input t-ref="spellcheck" type="text"/>
         <textarea t-ref="custom"/>
         <div t-ref="container">
           <input type="text" spellcheck="false"/>
           <div contenteditable="true"/>
         </div>
       </t>
```